Server/mapworker.cpp

```diff
@@ -50,8 +50,11 @@ void MapWorker::processItemPush(position pos, QTcpSocket *socket, playerMovement
     } while (type != itemType::furniture);
     position pushes = m_playerPositions[socket];
     position itemToPushCords=Utilities::getCoordsBySide(pushes,side);
-    QTcpSocket *itemToPush=getPlayerByPosition(itemToPushCords);
-    movePlayer(itemToPush, side);
+    position newPosItemtoPush=Utilities::getCoordsBySide(itemToPushCords,side);
+    m_itemController->addItem(Utilities::getCoordsBySide(itemToPushCords,side),id);
+    emit sendToAll("IPLACE:" + QByteArray::number(newPosItemtoPush.x) + ":" + QByteArray::number(newPosItemtoPush.y) + ":" + id + "|");
+    m_itemController->deleteItem(itemToPushCords,id);
+    emit sendToAll("ICLEAR:" + QByteArray::number(itemToPushCords.x) + ":" + QByteArray::number(itemToPushCords.y) + ":" + id + "|");
```

Comment:
54-57
First of all, please use "Clearing" functions e.g. we do not want to see few same items on map in the same time. Otherwise, players will have ability to create "cheat" which will be used for duping items.
At the moment, when server hit's 55th line we have two same objects: first on `itemToPushCords` and another on `newPosItemtoPush` cords. Please fix it.